testapp/views.py

```diff
@@ -0,0 +1,90 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+import transaction
+from sqlalchemy.exc import DBAPIError
+from .models import (
+    DBSession,
+    Entry,
+)
+from pyramid.httpexceptions import HTTPFound
+from jinja2 import Markup
+import markdown
+from testapp.formclass import EntryForm
+
+
+@view_config(route_name='new', renderer='templates/add.jinja2')
+def new_entry(request):
+    """Create a form page for a new entry."""
+    form = EntryForm(request.POST)
+    if request.method == 'POST' and form.validate():
+        new_entry = Entry(title=form.title.data, text=form.text.data)
+        DBSession.add(new_entry)
+        DBSession.flush()
+        url = request.route_url('entry', id=new_entry.id)
+        return HTTPFound(location=url)
+    return {'form': form}
+
+
+@view_config(route_name='edit', renderer='templates/add.jinja2')
+def edit_entry(request):
+    """Create a form page for an edited entry."""
+    edit_id = request.matchdict['id']
+    edit_entry = DBSession.query(Entry).get(edit_id)
+    # edit_entry.text = render_markdown(edit_entry.text) #trying to get edited entyr to not show html
+    form = EntryForm(request.POST, edit_entry)
+    if request.method == "POST" and form.validate():
+        # edit_entry.text = Markup(edit_entry.text)
+        # edit_entry.text = render_markdown(edit_entry.text)
+        # # edit_entry.text = Markup.striptags(edit_entry.text) #trying to get edited entyr to not show html
+        form.populate_obj(edit_entry)
+        DBSession.add(edit_entry)
+        DBSession.flush()
+        url = request.route_url('entry', id=edit_entry.id)
+        return HTTPFound(location=url)
+    return {'form': form}
```

Comment:
remove the dead code, but this is good.  The function does just what it should.  Your HTML problems are not arising in here.